core/utils/mailbox_prom.go

```diff
@@ -37,16 +38,21 @@ func (m *MailboxMonitor) Name() string { return "MailboxMonitor" }
 func (m *MailboxMonitor) Start(context.Context) error {
 	return m.StartOnce("MailboxMonitor", func() error {
 		t := time.NewTicker(WithJitter(mailboxPromInterval))
-		m.stop = t.Stop
-		// FIXME: monitorLoop never exits because t.Stop does not close the channel
-		go m.monitorLoop(t.C)
+		ctx, cancel := context.WithCancel(context.Background())
```

Comment:
The start context is only scoped to start-up and is not meant to be used for full app lifecycle. Each service's `Close()` method is called instead.

This used to be more important when we had network and DB calls inside of our `Start()` methods, but we've moved away from that in order to avoid blocking start-up, so it may be worth investigating if we can drop the context all together.